modules/unsupported/geojson-core/src/main/java/org/geotools/data/geojson/GeoJSONReader.java

```diff
@@ -358,9 +358,12 @@ public SimpleFeatureCollection getFeatures() throws IOException {
                 }
                 while (parser.nextToken() == JsonToken.START_OBJECT) {
                     ObjectNode node = mapper.readTree(parser);
-
-                    SimpleFeature feature = getNextFeature(node);
-                    features.add(feature);
+                    try {
+                        SimpleFeature feature = getNextFeature(node);
+                        features.add(feature);
+                    } catch (IOException e) {
+                        LOGGER.log(Level.WARNING, e.getMessage(), e);
+                    }
```

Comment:
And here... it's more tolerance for whatever other case. However, the typical data source in Geotools throws a straight exception when anything unexpected happens during a read. Wondering if there should be some flag to control tolerant reading vs fail fast, with fail fast being the default for consistency.

@ianturton opinions?

Reply:
i agree, i can undo this unless you want a flag.

Reply:
Normally data sources fail fast. Indeed the rendering subsystem is tolerant. 
Keeping in line with the other source behavior would be best in my mind.

Adding a Query parameter for tolerance would also be a good idea, but having just one implementation seems a bit odd. Do we have other cases where we can "resume" the read like here in JSON? Maybe in JDBC data store if the issue happens converting from resultset to feature, for example?

Reply:
I think that would be wise, some datastore may have non simple curves we cannot support like a spline.

Reply:
thanks for all the input and pointers!

i would like to remove this catch from the PR and only submit a fix for geometry=null. which is legal per spec so should be okay to change. it's also the original problem which i care about :) 

is that alright for you?

Reply:
Works for me, and @jodygarnett can take on adding new query parameters, and make their usage generic enough to be useful in 2-3 different places, control their activation, document them, and so on :laughing:

Reply:
okay! well, messy PR for such a simple change :grimacing: